server/backend/testspg/router/test_router_user_role_assignment.py

```diff
@@ -0,0 +1,418 @@
+import logging
+import copy
+import starlette.testclient
+from sqlalchemy.orm import Session
+from api.app.main import apiPrefix
+from api.app.jwt_validation import ERROR_PERMISSION_REQUIRED
+from api.app.crud import crud_application
+from api.app.crud import crud_user
+from api.app.crud import crud_role
+import testspg.jwt_utils as jwt_utils
+from testspg.constants import TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_CONCRETE, \
+    TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_ABSTRACT, \
+    TEST_USER_ROLE_ASSIGNMENT_FOM_TEST_CONCRETE, \
+    TEST_FOM_DEV_SUBMITTER_ROLE_ID, \
+    TEST_FOM_DEV_APPLICATION_ID, \
+    TEST_FOM_TEST_APPLICATION_ID
+
+LOGGER = logging.getLogger(__name__)
+endPoint = f"{apiPrefix}/user_role_assignment"
+
+FOM_DEV_ADMIN_ROLE = "FOM_DEV_ACCESS_ADMIN"
+FOM_TEST_ADMIN_ROLE = "FOM_TEST_ACCESS_ADMIN"
+ERROR_DUPLICATE_USER_ROLE = "Role already assigned to user."
+SUCCESS_RESPONSE = 200
+SUCCESS_RESPONSE_DELETE = 204
```

Comment:
Not sure if these http status codes are from previous written tests, we could use Python's http module `from http import HTTPStatus` for 200 and 204.

Reply:
yes!! updated